src/Microsoft.DotNet.Interactive.Jupyter/Messaging/Comms/ICommTarget.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Interactive.Jupyter.Messaging.Comms;
+
+internal interface ICommTarget
+{
+    public string Name { get; }
+
+    public void OnCommOpen(CommAgent commAgent, object data);
```

Comment:
What kinds of things can be passed to this `object data` parameter?

Reply:
this actually maps to the data returned by `comm_open` message and should match the Data property here. 
https://github.com/dotnet/interactive/blob/51c33c6ec96b3d6a468b5b4d603a472f69473611/src/Microsoft.DotNet.Interactive.Jupyter/Protocol/CommOpen.cs#L20

But this could be more specifically created as a dictionary of objects instead as defined by https://jupyter-client.readthedocs.io/en/stable/messaging.html#opening-a-comm. 

The dictionary could contain anything related to the target it's trying to open. I left it as object because there is a possibility that a kernel might not return a proper dictionary on comm_open. I could revisit this in a separate PR